### PR TITLE
Feature/3809-streaming-empty-form

### LINKED ISSUE #3809: Expenditures: As a user I can see an empty settings form for streaming payment
[Provided by Adrian via Contractors project management platform]

## Description
User has to be able to see an empty form for streaming payment. Start date should be pre-filled with current date.

## Acceptance Criteria

- an option to see a payment type
- an option to see to whom payment is streamed
- an option to see a starts date
- an option to see an ends date (by default when cancelled)

## Design

- [Figma Link - Streaming payment default](https://www.figma.com/file/dCtvv76S8mk5HNApjwmBto/Expenditures?node-id=5525%3A303773)

Default view

### PR DESCRIPTION
**Pull request by joannasia**
## Description

- User has to be able to see an empty form for streaming payment. Starts date should be pre-filled with current date.
- User has to be able to add recipient for streaming payment.
- User has to be able to choose how streaming payment should end. User has 3 options: "fixed time", "when canceled", "limit is reached".

**New stuff** ✨

* Streaming form with different fields.

**Changes** 🏗

* Changed ExpenditureSettings.tsx, ExpenditurePage/types.ts files

**TODO**

* Add datepicker.

Resolves #3809 #3812

### CODE REVIEW

**Review by willm30 — COMMENTED** [review]
I'm noticing a few discrepancies with the design. 
1. Hover state isn't as per: https://www.figma.com/file/dCtvv76S8mk5HNApjwmBto/Expenditures?node-id=5046%3A297336
2. Selected state isn't as per: https://www.figma.com/file/dCtvv76S8mk5HNApjwmBto/Expenditures?node-id=4882%3A299246
3. Date format / colour isn't as per design

**Comment by joannasia** [comment]
@willm30 point 1 and 2 are fixed - can you have a look please ?
point 3 - here is just a mock for future datepicker (i am waiting for the delivery of the component)

**Comment by agnieszkajarosikloj** [comment]
> @joannasia Looking good, thanks.
> 
> I have a question for either @agnieszkajarosikloj or @arrenv regarding the `Dropdown` component that's been created specifically for use in the `advanced-expenditures` feature.
> 
> Is this not a duplication of our core `Popover` component? It seems to do the same thing. Since it renders a portal outside of the `root` container, it also makes styling it using css (like how we generally do in the dapp) awkward. You could target the element like this: `div[class*="Dropdown"]`, but that selects every instance of it, so you can't target one instance specifically.
> 
> I don't remember the context for its creation so maybe I'm missing a use-case for the separate `Dropdown` component. But my initial sense is that it's duplicating code we already have. Thoughts welcome.

Dropdown component was created, because I needed element rendered not as a child, but outside the DOM hierarchy of the parent component, and also positioned relative to its parent.

**Comment by willm30** [comment]
> Dropdown component was created, because I needed element rendered not as a child, but outside the DOM hierarchy of the parent component, and also positioned relative to its parent.

@agnieszkajarosikloj Thanks for clarifying. I think it would be a good idea to leave a comment in the component explaining why it needs to be rendered outside of the dom hierarchy. That way, if someone else has the same question in the future, they won't be wondering.

**Comment by willm30** [comment]
@agnieszkajarosikloj Is the reason to do with the width of the element rendered in the dropdown?

File: src/modules/dashboard/components/ExpenditurePage/ExpenditureSettings/ExpenditureBaseForm.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import { COLONY_TOTAL_BALANCE_DOMAIN_ID } from '~constants';
+
+import BalanceSelect from './BalanceSelect';
+import { tokens as tokensData } from './constants';
+import styles from './ExpenditureSettings.css';
+import { Props } from './ExpenditureSettings';
+
+export const MSG = defineMessages({
+  team: {
+    id: 'dashboard.ExpenditurePage.ExpenditureBaseForm.team',
```
Comment (Julianahlk) [inline src/modules/dashboard/components/ExpenditurePage/ExpenditureSettings/ExpenditureBaseForm.tsx:20]: This and all ids below are missing .ExpenditureSettings, for example:`dashboard.ExpenditurePage.ExpenditureSettings.ExpenditureBaseForm.team`

File: src/modules/dashboard/components/ExpenditurePage/ExpenditureSettings/ExpenditureStreamingForm.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import { ExpenditureEndDateTypes } from '~pages/ExpenditurePage/types';
+
+import { supRenderAvatar } from '../Recipient/Recipient';
+
+import { Props } from './ExpenditureSettings';
+import styles from './ExpenditureSettings.css';
+
+export const MSG = defineMessages({
+  to: {
+    id: 'dashboard.ExpenditurePage.ExpenditureStreamingForm.to',
```
Comment (Julianahlk) [inline src/modules/dashboard/components/ExpenditurePage/ExpenditureSettings/ExpenditureStreamingForm.tsx:17]: Same here, this id and all ids below are missing .ExpenditureSettings: `dashboard.ExpenditurePage.ExpenditureSettings.ExpenditureStreamingForm.to`

**Review by Julianahlk — APPROVED** [review]
Besides the two comments I left, this PR is looking good from my side 👍

File: src/modules/dashboard/components/ExpenditurePage/ExpenditureSettings/ExpenditureBaseForm.tsx (hunk 1 above)
Comment (joannasia) [reply src/modules/dashboard/components/ExpenditurePage/ExpenditureSettings/ExpenditureBaseForm.tsx]: thanks! it is fixed

File: src/modules/dashboard/components/ExpenditurePage/ExpenditureSettings/ExpenditureStreamingForm.tsx (hunk 2 above)
Comment (joannasia) [reply src/modules/dashboard/components/ExpenditurePage/ExpenditureSettings/ExpenditureStreamingForm.tsx:17]: also it is fixed

**Comment by agnieszkajarosikloj** [comment]
@willm30 I've created Dropdown to fix this bug: https://github.com/JoinColony/colonyDapp/issues/3488

**Comment by joannasia** [comment]
@willm30 i have added the comment about the dropdown in `colonyDapp/src/modules/dashboard/components/ExpenditurePage/ExpenditureSettings/ExpenditureSettings.tsx`

**Comment by willm30** [comment]
@agnieszkajarosikloj Thank you for clarifying. 

@joannasia Would you mind putting the comment on line 24 of `dropdown.tsx` (the component itself), with a link to https://github.com/JoinColony/colonyDapp/issues/3488? This will help anyone coming to this in the future understand the problem it solves. Thanks!

**Comment by joannasia** [comment]
@willm30 i have added the comment, you can check here https://github.com/JoinColony/colonyDapp/pull/3833/files#diff-c0c03079472a668fa2f6707059099148d678a9ba6fcd31be614222a2f314341e

File: src/modules/dashboard/components/ExpenditurePage/ExpenditureSettings/ExpenditureBaseForm.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+export const MSG = defineMessages({
+  team: {
+    id:
+      'dashboard.ExpenditurePage.ExpenditureSettings.ExpenditureBaseForm.team',
+    defaultMessage: 'Team',
+  },
+  balance: {
+    id:
+      // eslint-disable-next-line max-len
+      'dashboard.ExpenditurePage.ExpenditureSettings.ExpenditureBaseForm.balance',
```
Comment (willm30) [inline src/modules/dashboard/components/ExpenditurePage/ExpenditureSettings/ExpenditureBaseForm.tsx:27]: Here you can put it in backticks `` and then you don't need the disable comment.

File: src/modules/dashboard/components/ExpenditurePage/ExpenditureSettings/ExpenditureSettings.tsx [hunk 4]
```diff
@@ -76,122 +66,40 @@ const expeditureTypes = [
...
-
-  const [activeToken, ...tokens] = tokensData;
+  const { values } = useFormikContext<ValuesType>() || {};
+  const expenditureType = values.expenditure;
 
   return (
     <div className={styles.container}>
       <FormSection appearance={{ border: 'bottom' }}>
         <div className={styles.blue}>
+          {/* withDropdownElement - dropdown component created because there is a need to render element outside DOM hierarchy */}
```
Comment (willm30) [inline src/modules/dashboard/components/ExpenditurePage/ExpenditureSettings/ExpenditureSettings.tsx:82]: This comment no longer necessary

**Review by willm30 — APPROVED** [review]
Looking good. Just a couple of comments.

File: src/modules/dashboard/components/ExpenditurePage/ExpenditureSettings/ExpenditureSettings.tsx (hunk 4 above)
Comment (joannasia) [reply src/modules/dashboard/components/ExpenditurePage/ExpenditureSettings/ExpenditureSettings.tsx]: sure, it is removed (https://github.com/JoinColony/colonyDapp/pull/3833/commits/7bb5d7d9edd766d43626010ec791d5c3bcd43002)

**Pull request merged**